### ISSUE TITLE
Detect bare `raise` outside of `except` block

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1974]
# Rule request

## Thesis

Report `raise` statements outside of `except` block

```python
# bad
def smth():
    raise

# good
def smth():
    try:
        ...
    except:
        raise
```

## Reasoning

It causes runtime error:

```
>>> smth()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in smth
RuntimeError: No active exception to reraise
```

It is possible that the function `smth` is meant to be called only from `except` block and so it won't fail but I'd say it is quite obscure and is a bad practice.

**Comment by sobolevn** [comment]
Ok, we used to detect this one, I can surely remember it. But for some reason, it is now gone.

**Comment by amansr02** [comment]
Hi. We are first-time contributors (@bhargavi247) and were wondering if we could take a look at this issue if possible? Do let us know if we can work on it. Thank you!

**Comment by sobolevn** [comment]
Sure, @amansr02! Thanks a lot for your help!

**Comment by bhargavi247** [comment]
Hello @sobolevn ! @amansr02 and I are making good progress! Just a quick question - we added a new violation that is called with an illegal bare raise. Are we supposed to add a new violation or use an existing violation?

**Comment by sobolevn** [comment]
Please, add a new one.

**Comment by danielboghean** [comment]
@sobolevn I'm taking a look at this issue and was able to add a new violation and visitor, but I have a few clarifying questions.

Is the intent to disallow any calls to `raise` unless it is from an `except` block? What about an `if` block? Something like 
```
if x < 0: 
    raise Exception("Sorry, no numbers below zero")
```

Are there any other blocks that `raise` should be allow to be called from?

Second, I ran my first implementation of the visitor, and there are a few files within the repo that are returning a bare raise violation. For example, it's finding this function as a violation since the raise is being called outside of an except or if clause:

```
def _convert_num(node: Optional[ast.AST]):
    if isinstance(node, Constant):  # pragma: py-lt-38
        if isinstance(node.value, (int, float, complex)):
            return node.value
    elif isinstance(node, ast.Num):  # pragma: py-gte-38
        return node.n
    # That's what is modified from the original
    elif isinstance(node, ast.Name):
        # We return string names as is, see how we return strings:
        return node.id
    raise ValueError('malformed node or string: {0!r}'.format(node))
```

There are 3 files currently where this is happening. How do we deal with this?

Thanks!

**Comment by sobolevn** [comment]
Hi, @danielboghean 

We only need to forbid bare `raise` without any arguments outside of `except` 🙂

**Comment by danielboghean** [comment]
Ah, I understand now, I totally misread the 'bare' except. Thanks for clearing that up, that should help!
